scim/data_service_principal.go

```diff
@@ -24,15 +24,31 @@ func DataSourceServicePrincipal() *schema.Resource {
 	return common.DataResource(spnData{}, func(ctx context.Context, e any, c *common.DatabricksClient) error {
 		response := e.(*spnData)
 		spnAPI := NewServicePrincipalsAPI(ctx, c)
-		spList, err := spnAPI.Filter(fmt.Sprintf("applicationId eq '%s'", response.ApplicationID), true)
+		var spList []User
+		var err error
+		if response.ApplicationID != "" {
+			spList, err = spnAPI.Filter(fmt.Sprintf("applicationId eq '%s'", response.ApplicationID), true)
```

Comment:
In case both of them are used, then we prefer the application id. Should we specify this in docs? Also should we throw error if application id doesn't match with display name?

Reply:
ah, yes - we need to throw an error...

Reply:
@tanmay-db fixed

Reply:
Thank you